### PR TITLE
Add mender params

### PR DESCRIPTION
**Pull request by mdanil007**

### CODE REVIEW

File: dominion/tasks.py [hunk 1]
```diff
@@ -184,8 +199,21 @@ def build(user_id, image):
...
+            'MENDER_ARTIFACT_NAME',
+            'MENDER_DATA_SIZE',
+            'MENDER_INVENTORY_POLL_INTERVAL',
+            'MENDER_TENANT_TOKEN',
+            'MENDER_RETRY_POLL_INTERVAL',
+            'MENDER_SERVER_URL',
+            'MENDER_UPDATE_POLL_INTERVAL'
         ]
         env.update({k: v for k, v in configuration.items() if k in allowed})
+        if (int(env['IMAGE_ROOTFS_SIZE']) > MAX_ALLOWED_ROOTFS_SIZE or int(env['MENDER_DATA_SIZE']) > MAX_ALLOWED_ROOTFS_SIZE):
```
Comment (denismosolov) [inline dominion/tasks.py]: Cannot be less than zero

File: dominion/tasks.py [hunk 2]
```diff
@@ -184,8 +199,21 @@ def build(user_id, image):
         allowed = [
             'HOST_NAME',
             'TIME_ZONE',
+            'IMAGE_ROOTFS_SIZE',
+            'ENABLEREDUCE',
```
Comment (denismosolov) [inline dominion/tasks.py]: We don't have `ENABLEREDUCE` in Pieman, let's remove it from here

File: dominion/tasks.py [hunk 3]
```diff
@@ -184,8 +199,21 @@ def build(user_id, image):
...
+            'MENDER_ARTIFACT_NAME',
+            'MENDER_DATA_SIZE',
+            'MENDER_INVENTORY_POLL_INTERVAL',
```
Comment (denismosolov) [inline dominion/tasks.py:205]: Let's check that `MENDER_INVENTORY_POLL_INTERVAL`, `MENDER_RETRY_POLL_INTERVAL` and `MENDER_UPDATE_POLL_INTERVAL` are greater than zero

**Review by denismosolov — CHANGES_REQUESTED** [review]

**Comment by denismosolov** [comment]
Got the following error when tried to build an artifact
```
[E 181204 11:58:59 server:116] an error occurred while executing the function
    Traceback (most recent call last):
      File "/home/denis/Documents/git/cusdeb/shirow/shirow/server.py", line 109, in _call_remote_procedure
        result = yield from future
      File "/usr/lib/python3.5/asyncio/futures.py", line 363, in __iter__
        return self.result()  # May raise too.
      File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
        raise self._exception
      File "/home/denis/Documents/git/cusdeb/blackmagic-env/lib/python3.5/site-packages/tornado/gen.py", line 292, in wrapper
        result = func(*args, **kwargs)
      File "/home/denis/Documents/git/cusdeb/blackmagic-env/lib/python3.5/types.py", line 243, in wrapped
        coro = func(*args, **kwargs)
      File "/home/denis/Documents/git/cusdeb/shirow/shirow/server.py", line 58, in wrapper
        return func(self, *args)
      File "bin/blackmagic.py", line 554, in search
        search=query)
      File "/home/denis/Documents/git/cusdeb/blackmagic-env/lib/python3.5/site-packages/pymongo/database.py", line 516, in command
        codec_options, **kwargs)
      File "/home/denis/Documents/git/cusdeb/blackmagic-env/lib/python3.5/site-packages/pymongo/database.py", line 428, in _command
        parse_write_concern_error=parse_write_concern_error)
      File "/home/denis/Documents/git/cusdeb/blackmagic-env/lib/python3.5/site-packages/pymongo/pool.py", line 477, in command
        collation=collation)
      File "/home/denis/Documents/git/cusdeb/blackmagic-env/lib/python3.5/site-packages/pymongo/network.py", line 116, in command
        parse_write_concern_error=parse_write_concern_error)
      File "/home/denis/Documents/git/cusdeb/blackmagic-env/lib/python3.5/site-packages/pymongo/helpers.py", line 210, in _check_command_response
        raise OperationFailure(msg % errmsg, code, response)
    pymongo.errors.OperationFailure: no such command: 'text', bad cmd: '{ text: "raspbian-stretch-armhf", search: "less" }'
[D 181204 12:00:00 blackmagic:241] Remove /home/denis/Documents/git/cusdeb/blackmagic-workspace/e2033ebc-6eff-478b-8e2e-9db4195abfe0
[E 181204 12:00:00 blackmagic:389] An exception was raised while building image
    Traceback (most recent call last):
      File "bin/blackmagic.py", line 387, in build
        ret_code = result.get()
      File "/home/denis/Documents/git/cusdeb/blackmagic-env/lib/python3.5/site-packages/celery/result.py", line 162, in get
        self.maybe_reraise()
      File "/home/denis/Documents/git/cusdeb/blackmagic-env/lib/python3.5/site-packages/celery/result.py", line 271, in maybe_reraise
        raise self.result
    KeyError: 'IMAGE_ROOTFS_SIZE'

```

File: dominion/tasks.py [hunk 4]
```diff
@@ -184,8 +199,40 @@ def build(user_id, image):
...
+            'MENDER_TENANT_TOKEN',
+            'MENDER_RETRY_POLL_INTERVAL',
+            'MENDER_SERVER_URL',
+            'MENDER_UPDATE_POLL_INTERVAL'
         ]
         env.update({k: v for k, v in configuration.items() if k in allowed})
+        if (int(env['MENDER_DATA_SIZE']) > MAX_ALLOWED_ROOTFS_SIZE or int(env['MENDER_DATA_SIZE']) > 0):
```
Comment (denismosolov) [inline dominion/tasks.py]:
```suggestion
        if (int(env['MENDER_DATA_SIZE']) > MAX_ALLOWED_ROOTFS_SIZE or int(env['MENDER_DATA_SIZE']) < 0):
```

**Review by denismosolov — CHANGES_REQUESTED** [review]

**Comment by denismosolov** [comment]
Need to validate only existing keys to prevent the following issue:

```
[2018-12-06 12:27:40,085: ERROR/MainProcess] Task tasks.build[da51acea-076c-4430-bb39-731bfe36dd5d] raised unexpected: KeyError('MENDER_DATA_SIZE',)
Traceback (most recent call last):
  File "/home/denis/Documents/git/cusdeb/dominion-env/lib/python3.5/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/denis/Documents/git/cusdeb/dominion-env/lib/python3.5/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/home/denis/Documents/git/cusdeb/dominion/dominion/tasks.py", line 212, in build
    if (int(env['MENDER_DATA_SIZE']) > MAX_ALLOWED_ROOTFS_SIZE or int(env['MENDER_DATA_SIZE']) > 0):
KeyError: 'MENDER_DATA_SIZE'
```

File: dominion/tasks.py [hunk 5]
```diff
@@ -184,8 +199,40 @@ def build(user_id, image):
...
+            routines.notify_us_on_fail(user_id, image)
+            routines.notify_user_on_fail(user, image)
+            return BUILD_FAILED
+
+        if (int(env['IMAGE_ROOTFS_SIZE']) > MAX_ALLOWED_ROOTFS_SIZE or int(env['IMAGE_ROOTFS_SIZE']) < 0):
+            routines.notify_us_on_fail(user_id, image)
+            routines.notify_user_on_fail(user, image)
+            return BUILD_FAILED
+
+        if (int(['MENDER_INVENTORY_POLL_INTERVAL']) < 0):
```
Comment (eugulixes) [inline dominion/tasks.py]: Missing `env`.

File: dominion/tasks.py [hunk 6]
```diff
@@ -184,8 +199,40 @@ def build(user_id, image):
...
+            routines.notify_us_on_fail(user_id, image)
+            routines.notify_user_on_fail(user, image)
+            return BUILD_FAILED
+
+        if (int(['MENDER_RETRY_POLL_INTERVAL']) < 0):
```
Comment (eugulixes) [inline dominion/tasks.py]: Missing `env`.

File: dominion/tasks.py [hunk 7]
```diff
@@ -184,8 +199,40 @@ def build(user_id, image):
...
+            routines.notify_us_on_fail(user_id, image)
+            routines.notify_user_on_fail(user, image)
+            return BUILD_FAILED
+
+        if (int(['MENDER_UPDATE_POLL_INTERVAL']) < 0):
```
Comment (eugulixes) [inline dominion/tasks.py]: Missing `env`.

**Review by denismosolov — DISMISSED** [review]

**Review by eugulixes — APPROVED** [review]

**Pull request merged**